opengemini_client/client_impl.py

```diff
@@ -234,3 +235,56 @@ def drop_retention_policy(self, dbname, retention_policy: str):
 
         query_string = f"DROP RETENTION POLICY {retention_policy} ON {dbname}"
         return self._query_post(Query(database=dbname, command=query_string, retention_policy=retention_policy))
+
+    def _show_query_result(self, database, command: str) -> QueryResult:
+        if not database:
+            raise ValueError("empty database name")
+        if not command:
+            raise ValueError("empty query command")
+        return self.query(query=Query(database=database, command=command, retention_policy=''))
+
+    def _show_with_result_any(self, database, command: str) -> List[ValuesResult]:
+        values_results = []
+        query_results = self._show_query_result(database, command)
+        if len(query_results.results) == 0:
+            return values_results
+        for res in query_results.results[0].series:
+            values_result = ValuesResult(measurement=res.name, values=[])
+            for values in res.values:
+                for value in values:
+                    values_result.values.append(value)
+            values_results.append(values_result)
+        return values_results
+
+    def _show_with_result_key_value(self, database, command: str) -> List[ValuesResult]:
+        values_results = []
+        query_results = self._show_query_result(database, command)
+        if len(query_results.results) == 0:
+            return values_results
+        for res in query_results.results[0].series:
+            values_result = ValuesResult(measurement=res.name, values=[])
+            for values in res.values:
+                if len(values) < 2:
+                    continue
+                values_result.values.append(KeyValue(name=values[0], value=values[1]))
+            values_results.append(values_result)
+        return values_results
+
+    def show_tag_keys(self, database, command: str) -> List[ValuesResult]:
+        return self._show_with_result_any(database, command)
+
+    def show_tag_values(self, database, command: str) -> List[ValuesResult]:
+        return self._show_with_result_key_value(database, command)
+
+    def show_field_keys(self, database, command: str) -> List[ValuesResult]:
+        return self._show_with_result_key_value(database, command)
+
+    def show_series(self, database, command: str) -> List[str]:
+        values_result = self._show_with_result_any(database, command)
+        series = []
+        if len(values_result) == 0:
+            return series
+        for value in values_result[0].values:
+            if isinstance(value, str):
```

Comment:
Don't we need to handle types other than str here?

Reply:
results don't have other types than str
![image](https://github.com/openGemini/opengemini-client-python/assets/55304072/97514ca7-ccc8-4c82-be49-3429963f66e7)